src/Routes/RosPage/RosPage.js

```diff
@@ -92,6 +95,30 @@ class RosPage extends React.Component {
         }
     }
 
+    processOsVersion() {
+        let osObject = {};
+        osObject.label = 'Operating system';
+        osObject.type = 'checkbox';
+        osObject.filterValues = {};
+        this.fetchSystems({
+            perPage: -1
+        }).then((response) => {
+            osObject.filterValues.items = Array.from(new Set((response.data).reduce((filtered, system) => {
+                if (system.os) {
+                    filtered.push(system.os);
+                }
+
+                return filtered;
```

Comment:
What you think about putting this part into a function where we extract OS versions from the response to improve readability?

Reply:
I extracted this part from `componentDidMount` function. This function mainly has logic related to the OS version. Extracting again would not make sense, if not, we can change the function name to `getOSversionObject` or something like that. What's your thought on this?

Reply:
okay let's keep it as it s for now, if needed we will refactor later 💯